src/ray/gcs/redis_context.cc

```diff
@@ -24,24 +24,37 @@ void GlobalRedisCallback(void *c, void *r, void *privdata) {
   }
   int64_t callback_index = reinterpret_cast<int64_t>(privdata);
   redisReply *reply = reinterpret_cast<redisReply *>(r);
+  // We use an optional response to distinguish between a nil response and a
+  // response of an empty string.
+  boost::optional<const std::string &> optional_data;
```

Comment:
Please don't use boost optional.  My unpopular opinion is we shouldn't use boost at all.

boost::optional is not in the approved subset by Google style: http://google.github.io/styleguide/cppguide.html#Boost

We should transition to abseil.  https://abseil.io/tips/123

Reply:
+1 to not using boost anywhere except for asio.